p2p/src/sessions/mod.rs

```diff
@@ -372,3 +423,59 @@ pub struct GetConsolidatedPeersResult {
     /// The address shown here can be used to connect to this peers in the future.
     pub outbound: Vec<SocketAddr>,
 }
+
+/// A convenient wrapper around a collection of network ranges.
+/// This enables efficient tracking of the inbound connections we have so as to prevent sybil
+/// machines from monopolizing our inbound peers table.
+#[derive(Default)]
+pub struct NetworkRangesCollection {
+    inner: BTreeSet<Vec<u8>>,
```

Comment:
This would be much more efficient if we know the size of the ip range and use a fixed sized array instead. In ipv4 it's 2 bytes but in ipv6 it's 8 bytes. Since I'm not sure if we currently support ipv6, how about setting it to 2 bytes:

```suggestion
    inner: BTreeSet<[u8; 2]>,
```

And we can use

```rust
let (_, range, _) = split_socket_addresses(address);
let range = [range[0], range[1]];
```